examples/scripts/learning_regularization_parameter_maps_for_tv_reconstruction.py

```diff
@@ -0,0 +1,876 @@
+            )
+        axes[0, i].set_axis_off()
+    plt.tight_layout()
+    plt.show()
+
+
+# %% [markdown]
+# Let us have a look at one example of the BrainWeb images.
+
+# %%
+
+image_ = next(iter(dataloader_validation))[0]
+
+n_coils = 8
+noise_variance = 0.05
+acceleration_factor = 4
+kdata_, csm_, traj_, adjoint_recon_, pseudo_inverse_solution_, image_ = dataset_validation.prepare_data(
+    image_, n_coils=n_coils, acceleration_factor=acceleration_factor, noise_variance=noise_variance
+)
+
+show_images(
+    adjoint_recon_.squeeze(),
+    pseudo_inverse_solution_.squeeze(),
+    image_.squeeze(),
+    titles=['Adjoint', 'Pseudo-Inverse', 'Target'],
+    rotate=False,
+    show_mse=True,
+    clim=(0, 0.5),
+)
+
+# %% [markdown]
+# ## Network training
+
+# %% [markdown]
+# We now define the unrolled PDHG network by instantiating the `AdaptiveTVNetwork2D` class.
+
+# %%
+torch.manual_seed(2025)
+lambda_map_network = ParameterMapNetwork2D(n_filters=32)
+n_iterations = 128
+
+img_shape = mrpro.data.SpatialDimension(z=1, y=images.shape[-2], x=images.shape[-1])
+k_shape = mrpro.data.SpatialDimension(z=1, y=images.shape[-2], x=images.shape[-1])
+
+adaptive_tv_network = AdaptiveTVNetwork2D(
+    img_shape=img_shape,
+    k_shape=k_shape,
+    lambda_map_network=lambda_map_network,
+    n_iterations=n_iterations,
+)
+
+if torch.cuda.is_available():
+    adaptive_tv_network = adaptive_tv_network.cuda()
+    pseudo_inverse_solution_ = pseudo_inverse_solution_.cuda()
+    csm_ = csm_.cuda()
+    kdata_ = kdata_.cuda()
+    traj_ = traj_.cuda()
+
+with torch.no_grad():
+    regularization_parameter_map_init = adaptive_tv_network.estimate_lambda_map(pseudo_inverse_solution_)
+
+
+# %% [markdown]
+# Let us now train the unrolled PDHG network. We set up the optimizer with an appropriate learning rate, choose a
+# number of epochs and write a simple training loop, in which we present the network input-target pairs and update
+# the network's parameters by gradient-descent.
+# To obtain a somewhat decent estimate of the network parameters, training should take approximately 5-6 minutes
+# on a GPU. If you have GPU and you need a break, start the training now, grab yourself a coffee and come back later
+# to see your network trained from scratch.
+# If you do not have a GPU, we provide a pre-trained model, which you can load and directly apply to the data.
+
+# %%
+from typing import cast
+
+cnn_block = cast(torch.nn.Module, adaptive_tv_network.lambda_map_network.cnn_block)
+optimizer = torch.optim.Adam(
+    [
+        {'params': cnn_block.parameters(), 'lr': 1e-4},
+        {'params': [adaptive_tv_network.lambda_map_network.t], 'lr': 1e-1},
+    ],
+    weight_decay=1e-5,
+)
+
+import copy
+import datetime
+from time import time
+
+from tqdm import tqdm
+
+n_epochs = 10
+validation_loss_values = []
+best_model = None
+noise_variance = 0.05
+acceleration_factor = 4
+
+time_start = time()
+
+if torch.cuda.is_available():
```

Comment:
if not torch.cuda.is_available()
 # without a gpu, skip training
     n_epochs=0
     adaptive_tv_network.load_state_dict(torch.load(data_folder_tv / 'tv_model.pt', map_location='cpu'))